### PR TITLE
dydisnix: init at unstable

### PR DESCRIPTION
**Pull request by tomberek**
###### Motivation for this change
Include dydisnix into nixpkgs
@svanderburg: inclusion into nixpkgs may provide easier entry for additional users, thoughts?

###### Things done
Added `./bootstrap` to the standard configure/make/make install.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by svanderburg** [comment]
Interesting to see that there are people who want to use this package and integrate it into Nixpkgs!

But, I want to raise a big warning: dydisnix is by far not as mature as the core toolset: disnix. It should not be considered as reliable or usable. For example: it lacks the same quality level in terms of documentation and code compared to the core package. Originally, the package was developed for a research paper under enormous time pressure.

Moreover, dydisnix has no decent end-user documentation, aside from a simple README.md, and it has never been used for production deployments (at least not by me). 

Furthermore, I also have not made any commitments in providing stable interfaces of the APIs and command-line tools.

Although I'm occasionally working on Disnix and the extensions, including dydisnix, it still requires some work and stabilization effort to create an initial 0.1 release. This is why I have not created any releases yet!

I don't mind that this change gets integrated into Nixpkgs, but we somehow need to remind potential users that dydisnix is a highly experimental prototype that is subject to major changes!

File: pkgs/tools/package-management/disnix/dydisnix/default.nix
Comment (xeji) [inline pkgs/tools/package-management/disnix/dydisnix/default.nix]: 0.0 doesn't make much sense. When there's no tagged upstream release, we typically use something like `dydisnix-unstable-2018-04-27` (the date should be the date of the git commit referenced).

File: pkgs/tools/package-management/disnix/dydisnix/default.nix
Comment (xeji) [inline pkgs/tools/package-management/disnix/dydisnix/default.nix]: Maybe add a `longDescription` mentioning that this is experimental software and should not be used in production.

**Comment by svanderburg** [comment]
@Synthetica9 Technically, the package is not broken, just not officially released or production ready :)

File: pkgs/tools/package-management/disnix/dydisnix/default.nix
Comment (tomberek) [reply pkgs/tools/package-management/disnix/dydisnix/default.nix]: updated, thx

File: pkgs/tools/package-management/disnix/dydisnix/default.nix
Comment (tomberek) [reply pkgs/tools/package-management/disnix/dydisnix/default.nix]: added.

**Comment by Ekleog** [comment]
(triage) @svanderburg Do you prefer this PR merged, not merged, merged with `meta.broken = true;` or some fourth option? IMO all options would make sense, so we only need someone to decide -- here it'd make sense for this someone to be you, I think.

File: pkgs/tools/package-management/disnix/dydisnix/default.nix
Comment (ryantm) [inline pkgs/tools/package-management/disnix/dydisnix/default.nix]:
```suggestion
  pname = "dydisnix";
```

**Review by ryantm — CHANGES_REQUESTED** [review]

**Comment by ryantm** [comment]
@tomberek Based on my reading of these comments, it seems fine to merge this after fixing up the pname thing.

**Comment by ryantm** [comment]
@GrahamcOfBorg build dydisnix

**Review by ryantm — APPROVED** [review]

**Comment by ryantm** [comment]
@GrahamcOfBorg build dydisnix

**Comment by tomberek** [comment]
still important to me

**Review by ryantm — CHANGES_REQUESTED** [review]
Please fix build failures.

**Comment by roberth** [comment]
@tomberek your recent commits have an e-mail address that isn't recognized by GitHub. You might want to change those (`git config`, `git rebase -i`, rebase `edit` command, `git commit --reset-author`) or configure an additional email in https://github.com/settings/emails

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 49366` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>dydisnix</li>
    <li>libnixxml</li>
  </ul>
</details>

**Review by lukegb — APPROVED** [review]
Builds for me, all the binaries at least run and display their help.

File: pkgs/tools/package-management/disnix/default.nix
Comment (lukegb) [inline pkgs/tools/package-management/disnix/default.nix:18]: tiny nit: this and the dydisnix meta attrsets would be cleaner with `with stdenv.lib;` like you've done in libnixxml, but it shouldn't block merging this.

**Pull request merged**
